api/v1/sp.go

```diff
@@ -43,8 +50,19 @@ type StorageProviderIDs struct {
 //	@Produce		json
 //	@Success		200	{object}	ResponseEnvelope{response=GetStorageProvidersResponse}
 //	@Router			/sp/approve [post]
-func handleApproveStorageProviders(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+func (a *apiV1) handleApproveStorageProviders(c echo.Context) error {
+	var storageProviderIds Sps
+
+	err := json.NewDecoder(c.Request().Body).Decode(&storageProviderIds)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	var storageProviderResponse StorageProvidersResponse
+	a.db.Table("sps").Where("sp_id in (?)", storageProviderIds.SpId).UpdateColumn("tenant_sp_state", "eligible") //@jcace is active the right state
```

Comment:
There needs to be a bit more logic here,

It's only possible to approve a sp that is in `pending` state - if they are `eligible` or `suspended` (or already `active`, i guess), then this should be an error for that one.  So we need to check this

Also instead of string value `"eligible"` you can use the enum value i.e `db.TenantSpStateActive`  (add `	"github.com/data-preservation-programs/spade-tenant/db"` to imports at the top of the file)

Reply:
the more "gormy" way to do this should be like 

```	a.db.Model(&db.TenantsSPs{}).Where("sp_id in (?)", storageProviderIds.SpId).UpdateColumn("tenant_sp_state", db.TenantSpStateActive)```

btw - can you test and see if that one works?